src/main/java/pl/agh/edu/iisg/io/vmms/vmmsbackend/controller/ReservationController.java

```diff
@@ -0,0 +1,233 @@
+            @RequestParam("to") @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm") Date to) {
+        return reservationService
+                .getConfirmedOrBeforeDeadlineToConfirmReservationsBetweenDates(from, to)
+                .stream()
+                .filter(r -> r.getOwner().getId().equals(userId))
+                .map(this::convertToDto)
+                .collect(Collectors.toList());
+    }
+
+    @RequestMapping(path = "vm/{vmShortName}/between", method = RequestMethod.GET)
+    public List<ReservationDto> getReservationsBetweenDatesForVMPool(
+            @PathVariable("vmShortName") String vmPoolShortName,
+            @RequestParam("from") @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm") Date from,
+            @RequestParam("to") @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm") Date to) {
+        return reservationService
+                .getConfirmedOrBeforeDeadlineToConfirmReservationsBetweenDatesForVMPool(vmPoolShortName, from, to)
+                .stream()
+                .map(this::convertToDto)
+                .collect(Collectors.toList());
+    }
+
+    @RequestMapping(path = RESERVATION_ENDPOINT, method = RequestMethod.GET)
+    public ReservationDto getReservation(
+            @RequestParam("reservationId") Long id) throws HttpException {
+        Optional<Reservation> reservation = reservationService.find(id);
+        if (reservation.isPresent()) {
+            return convertToDto(reservation.get());
+        } else {
+            throw new ReservationExpiredException();
+        }
+    }
+
+    @RequestMapping(path = RESERVATION_ENDPOINT, method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.CREATED)
+    public ReservationResponseDto createReservation(
+            @Valid @RequestBody ReservationRequestDto reservationRequest) {
+        Reservation reservation = convertToReservation(reservationRequest);
+        reservation = reservationService.saveTemporary(
+                reservation,
+                reservationRequest.getStartTime(),
+                reservationRequest.getEndTime(),
+                reservationRequest.getDates());
+
+        List<Date> daysReserved = reservation
+                .getPeriods()
+                .stream()
+                .map(this::convertPeriodToDay)
+                .collect(Collectors.toList());
+
+        List<Date> daysRequested = reservationRequest.getDates();
+        daysRequested.removeAll(daysReserved);
+        return new ReservationResponseDto(reservation.getId(), daysRequested);
+    }
+
+    @RequestMapping(path = CONFIRM_TMP_ENDPOINT, method = RequestMethod.PUT)
+    public String confirmReservation(
+            @RequestBody Long reservationId) throws ReservationExpiredException {
+        Optional<Reservation> reservation = reservationService.findIfNotExpired(reservationId);
+        if (reservation.isPresent()) {
+            reservationService.confirm(reservation.get());
+            return "CONFIRMED";
```

Comment:
This should not be a string - make it a simple HTTP Response with 200 code or other more specific